lib/spack/llnl/util/filesystem.py

```diff
@@ -1816,66 +1823,67 @@ def find(root, files, recursive=True):
         files (str or collections.abc.Sequence): Library name(s) to search for
         recursive (bool): if False search only root folder,
             if True descends top-down from the root. Defaults to True.
+        max_depth (int): if set, don't search below this depth. Cannot be set
+            if recursive is False
 
     Returns:
         list: The files that have been found
     """
     if isinstance(files, str):
         files = [files]
 
-    if recursive:
-        tty.debug(f"Find (recursive): {root} {str(files)}")
-        result = _find_recursive(root, files)
-    else:
-        tty.debug(f"Find (not recursive): {root} {str(files)}")
-        result = _find_non_recursive(root, files)
+    if max_depth is not _unset and not recursive:
+        raise ValueError(f"max_depth ({max_depth}) cannot be set if recursive is False")
+
+    if not recursive:
+        max_depth = 0
+
+    tty.debug(f"Find (max depth = {max_depth}): {root} {str(files)}")
```

Comment:
Nitpick: I'd rather don't see `llnl/util/filesystem.py` use `tty.debug`. Preferably that only happens in `cmd/*.py` as those are command line entry points, whereas internals may be used in scripts where stdout output as a side effect is undesirable.

Reply:
This seems like a higher-level concern: this prints no more than was printed before (more or less just moves print statements, and keeps them `debug` as they were before). I think that merits its own discussion (outside of this PR) and a larger, more-focused refactor.

Reply:
Yes, not a blocker for this PR. Just weird if you use the Spack API that it has annoying side effects you can't disable.